### PR TITLE
feat(sparq-vectors): hybrid_search fusing nearest_term + most_similar by Term via RRF (sq-88c6)

### PR DESCRIPTION
**Pull request by jeswr**
> 🤖 **SPARQ agent** — implementing bead **sq-88c6** (sparq-vectors, P3): a hybrid-retrieval helper that fuses semantic similarity and ANN vector search by `Term` via Reciprocal Rank Fusion.

## What this adds

**`hybrid_search`** — a one-call hybrid-retrieval helper that drives **N retriever closures** off a single query item and fuses their ranked lists **by item** using the existing `fuse_rrf` (Reciprocal Rank Fusion, `score(t) = Σ 1/(k + rank_i(t))`, default `k = 60`). The canonical use fuses:

- this crate's embedding **ANN** — `VectorIndex::nearest_term` (cosine), and
- `sparq-sim`'s structural **semantic** similarity — `Sim::most_similar` (weighted Jaccard)

**by `Term`**. A term ranked high in *both* retrievers outranks one high in only one (consensus wins); a term in only one list still surfaces; results are **deduplicated by `Term`**.

```rust
let fused = hybrid_search(&query, 10, RRF_K, &mut [
    &mut |t: &Term| index.nearest_term(t, &graph, &store, 50)        // ANN (cosine, f32→f64)
        .into_iter().map(|(t, s)| (t, s as f64)).collect(),
    &mut |t: &Term| sim.most_similar(t, 50),                          // structural (Jaccard)
]);
```

## Design notes

- **`fuse_rrf` already existed** and is **reused unchanged** — no duplication. `hybrid_search` is a thin orchestration layer over it.
- The helper is **generic over retriever closures** (`Retriever<'_, Q, T> = &mut dyn FnMut(&Q) -> Vec<(T, f64)>`), so the **library stays decoupled** from `sparq-sim`. That crate is a **dev-dependency only** — used by the integration example/test, never entering the published dependency graph. This preserves the crate's deliberate *"neither crate depends on the other"* design (documented in `fuse.rs`).
- RRF ignores score *scales* (cosine in `[-1,1]` vs Jaccard in `[0,1]`) — only ranks matter; the `f32→f64` widen on `nearest_term` is a pure type bridge.

## Tests (all green)

- **5 unit tests** (`fuse.rs`): RRF consensus (high-in-both beats high-in-one), hand-computed scores, dedup, `top_k` truncation, edge cases (no / empty / all-empty retrievers, `top_k == 0`), term-in-one-list-only, identical-lists doubling.
- **2 integration tests** (`tests/hybrid.rs`): fuse the **real** `nearest_term` + `most_similar` over a built store + graph (consensus ordering + exact RRF score `2/(k+1)`), and unknown-query → empty.

## Gate

- `cargo test -p sparq-vectors` — 43 lib + 2 integration + doc-tests, all pass.
- `cargo clippy -p sparq-vectors --all-targets -- -D warnings` — clean.
- `skills/vector-search/SKILL.md` + crate `README.md` updated for the new public API (`hybrid_search`, `Retriever`).

Touches only `sparq-vectors` (+ its SKILL.md). References bead **sq-88c6**.

🤖 Generated with [Claude Code](https://claude.com/claude-code)

### CODE REVIEW

File: skills/vector-search/SKILL.md [hunk 1]
```diff
@@ -125,6 +125,9 @@ EncodedStore::rank_pq(&self, &DistanceTable, k) -> Vec<(Id, f32)>;  cosine_from_
 fuse_rrf(lists: &[&[(T, f64)]], k: f64 /*RRF_K=60.0*/, top_k) -> Vec<(T, f64)>
 fuse_rrf_weighted(lists: &[(&[(T, f64)], f64)], k, top_k) -> Vec<(T, f64)>      // weight 0.0 mutes a list entirely
 fuse_scores(a: &[(T,f64)], b: &[(T,f64)], alpha /*1.0=a only*/, top_k) -> Vec<(T, f64)>
+// one-call hybrid: run N retriever closures on one query, fuse by item via RRF, dedup
+hybrid_search(query: &Q, top_k, k /*RRF_K*/, &mut [Retriever<Q,T>]) -> Vec<(T, f64)>
+//   Retriever<'_, Q, T> = &mut dyn FnMut(&Q) -> Vec<(T, f64)>  (e.g. nearest_term / most_similar closures)
```
[the remark this answers is not in the document]
Comment (jeswr) [reply skills/vector-search/SKILL.md]: > 🤖 **SPARQ agent** — Fixed in 1f02e87f62a9f14eb35ffa3bcf1b1e5a00f52798. The cheat-sheet now reads `&mut [Retriever<'_, Q, T>]` and the alias line shows the lifetime: `Retriever<'r, Q, T> = &'r mut dyn FnMut(&Q) -> Vec<(T, f64)>`, matching the public definition in `crates/sparq-vectors/src/fuse.rs:131`. Good catch — `Retriever<Q, T>` was not a valid type as written.

File: skills/vector-search/SKILL.md [hunk 2]
```diff
@@ -239,6 +242,20 @@ let hybrid  = fuse_rrf(&[&text, &structural], RRF_K, 10);   // rank-only; right
...
+`Term` and fuse by `Term` in a single call (dedups; a term in only one list still surfaces;
+RRF ignores the score scales). Widen `nearest_term`'s `f32` to `f64` so both lists share the
+`(Term, f64)` shape:
+
+```rust
+use sparq_vectors::{hybrid_search, RRF_K};
+let fused = hybrid_search(&query, 10, RRF_K, &mut [
+    &mut |t: &oxrdf::Term| index.nearest_term(t, &graph, &store, 50)
+        .into_iter().map(|(t, s)| (t, s as f64)).collect(),   // ANN (cosine)
+    &mut |t: &oxrdf::Term| Sim::new(&graph).most_similar(t, 50),  // structural (Jaccard)
```
[the remark this answers is not in the document]
Comment (jeswr) [reply skills/vector-search/SKILL.md]: > 🤖 **SPARQ agent** — Fixed in 1f02e87f62a9f14eb35ffa3bcf1b1e5a00f52798. The standalone `hybrid_search` snippet now calls `sparq_sim::Sim::new(&graph)` via the fully-qualified path, so the block is self-contained and no longer depends on the `use sparq_sim::Sim;` from the earlier code block.
